src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,105 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
 public class HelloWorldTest {
 
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        // Clear storage before each test
+        core.basesyntax.db.Storage.people.clear();
+    }
+
+    @Test
+    public void register_validUser_userRegistered() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
+    public void register_existingLogin_userNotRegistered() {
+        User user1 = new User();
+        user1.setLogin("existingLogin");
+        user1.setPassword("password1");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("existingLogin");
+        user2.setPassword("password2");
+        user2.setAge(25);
+
+        InvalidUserException thrown = assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user2);
+        });
+        assertEquals("User with this login already exists", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullUser_throwsException() {
```

Comment:
The test case name `register_nullUser_throwsException` is good as it follows the naming convention. However, ensure that all test methods follow this pattern for consistency.